contracts/interchain-token-service/src/state.rs

```diff
@@ -126,10 +126,19 @@ pub struct TokenConfig {
     pub origin_chain: ChainNameRaw,
 }
 
+type TokenAddress = nonempty::HexBinary;
+#[cw_serde]
+pub struct CustomToken {
```

Comment:
```suggestion
pub struct TokenMetadata {
```
Custom feels weird, this can be metadata for any token

Reply:
NonItsToken? I want it to be clear that this is a special case

Reply:
yeah this is a special case, it can't be metadata for any token

Reply:
I don't follow, this could be any token on that chain. Even ITS tokens adhere to the chain's token standard and could be linked as a custom token.

This functionality allows registering metadata for any existing token (whether it's a standard ERC20, ITS ERC20 token, custom token or some other standard doesn't matter to ITS Hub)

Reply:
hmm yes i guess that's true (though I think this sort of behavior is very hard to reason about and we should probably find a way to improve this aspect of ITS). But the issue is that not all tokens will have metadata stored in this struct. So we need some way to reflect that special case

Reply:
I couldn't think of a better name (don't like `TokenMetadata`), so I just left it as is for now, we can change it later if we want